src/d/actor/d_a_obj_doguu_demo.cpp

```diff
@@ -4,50 +4,145 @@
 //
 
 #include "d/actor/d_a_obj_doguu_demo.h"
+#include "d/d_bg_w.h"
+#include "d/d_cc_d.h"
+#include "d/d_com_inf_game.h"
+#include "d/d_procname.h"
 #include "dolphin/types.h"
+#include "f_op/f_op_actor.h"
+#include "f_op/f_op_actor_mng.h"
+#include "m_Do/m_Do_mtx.h"
 
-/* 00000078-00000098       .text CheckCreateHeap__FP10fopAc_ac_c */
-static BOOL CheckCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL CheckCreateHeap(fopAc_ac_c* i_this) {
+    return ((daObjDoguuD_c*)i_this)->CreateHeap();
 }
 
-/* 00000098-00000214       .text CreateHeap__13daObjDoguuD_cFv */
-void daObjDoguuD_c::CreateHeap() {
-    /* Nonmatching */
+BOOL daObjDoguuD_c::CreateHeap() {
+    this->param = (u8)mBase.mParameters;  // Im not sure why this is here, none of the other classes
+                                          // I looked at did this
+
+    J3DModelData* modelData = (J3DModelData*)(dComIfG_getObjectRes("DoguuD", 0x04));
+    JUT_ASSERT(0x65, modelData != 0);
+    mpModel = mDoExt_J3DModel__create(modelData, 0x00, 0x11020203);
+    if (mpModel == NULL)
+        return FALSE;
+
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::YrotM(shape_angle.y);
+    mDoMtx_stack_c::scaleM(mScale);
+    mDoMtx_copy(mDoMtx_stack_c::get(), mtx);
+
+    mpBgW = new dBgW();
+
+    if (mpBgW == NULL) {
+        return FALSE;
+    }
+
+    if (mpBgW->Set((cBgD_t*)dComIfG_getObjectRes("DoguuD", 0x07), cBgW::MOVE_BG_e, &mtx))
+        return FALSE;
+
+    return TRUE;
 }
 
-/* 00000214-00000294       .text CreateInit__13daObjDoguuD_cFv */
 void daObjDoguuD_c::CreateInit() {
-    /* Nonmatching */
+    fopAcM_SetMtx(this, mpModel->getBaseTRMtx());
+    fopAcM_setCullSizeBox(this, -30000.0, -5000.0, -30000.0, 30000.0, 40000.0, 30000.0);
```

Comment:
```suggestion
    fopAcM_setCullSizeBox(this, -30000.0f, -5000.0f, -30000.0f, 30000.0f, 40000.0f, 30000.0f);
```
single precision floats should be suffixed with `f` to distinguish from double precision floats